zebrad/src/commands/start.rs

```diff
@@ -1,46 +1,101 @@
-//! `start` subcommand - example of how to write a subcommand
-
-/// App-local prelude includes `app_reader()`/`app_writer()`/`app_config()`
-/// accessors along with logging macros. Customize as you see fit.
-use crate::prelude::*;
-
+//! `start` subcommand - entry point for starting a zebra node
+//!
+//!  ## Application Structure
+//!
+//!  A zebra node consists of the following services and tasks:
+//!
+//!  * Network Service
+//!    * primary interface to the node
+//!    * handles all external network requests for the zcash protocol
+//!      * via zebra_network::Message and zebra_network::Response
+//!    * provides an interface to the rest of the network for other services and
+//!    tasks running within this node
+//!      * via zebra_network::Request
+//!  * Consensus Service
+//!    * handles all validation logic for the node
+//!    * verifies blocks using zebra-chain and zebra-script, then stores verified
+//!    blocks in zebra-state
+//!  * Sync Task
+//!    * This task runs in the background and continouously queries the network for
+//!    new blocks to be verified and added to the local state
 use crate::config::ZebradConfig;
-
+use crate::{components::tokio::TokioComponent, prelude::*};
 use abscissa_core::{config, Command, FrameworkError, Options, Runnable};
+use color_eyre::Report;
+use eyre::{eyre, WrapErr};
+use futures::{
+    prelude::*,
+    stream::{FuturesUnordered, StreamExt},
+};
+use std::collections::BTreeSet;
+use tower::{buffer::Buffer, service_fn, Service, ServiceExt};
+use zebra_chain::{block::BlockHeaderHash, types::BlockHeight};
 
-use futures::prelude::*;
+// genesis
+static GENESIS: BlockHeaderHash = BlockHeaderHash([
+    8, 206, 61, 151, 49, 176, 0, 192, 131, 56, 69, 92, 138, 74, 107, 208, 93, 161, 110, 38, 177,
+    29, 170, 27, 145, 113, 132, 236, 232, 15, 4, 0,
+]);
 
 /// `start` subcommand
-///
-/// The `Options` proc macro generates an option parser based on the struct
-/// definition, and is defined in the `gumdrop` crate. See their documentation
-/// for a more comprehensive example:
-///
-/// <https://docs.rs/gumdrop/>
 #[derive(Command, Debug, Options)]
-pub struct StartCmd {
+pub struct StartArgs {
     /// Filter strings
     #[options(free)]
     filters: Vec<String>,
+
+    /// The address of the node to connect to.
+    #[options(
+        help = "The address of the node to connect to.",
+        default = "127.0.0.1:8233"
+    )]
+    addr: std::net::SocketAddr,
```

Comment:
I think we don't want to carry this code over, because rather than connecting to a specific peer we want to connect to the network generally, so we don't need to override the provided list of peers.

Reply:
And we have our own seeders now too 🌱